apps/transport/lib/transport/stat_handler.ex

```diff
@@ -0,0 +1,213 @@
+          }
+        )
+      )
+
+    aoms_with_datasets = aoms |> Enum.filter(&(&1.nb_datasets > 0 || !is_nil(&1.parent_dataset_id)))
+
+    regions = Repo.all(from(r in Region, where: r.nom != "National"))
+
+    aoms_max_severity = compute_aom_max_severity()
+
+    total_aom_with_datasets =
+      aoms_max_severity
+      |> Map.values()
+      |> Enum.filter(fn error -> not is_nil(error) end)
+      |> Enum.sum()
+
+    %{
+      nb_datasets: Repo.aggregate(Dataset, :count, :id),
+      nb_pt_datasets: Dataset.count_by_type("public-transit"),
+      nb_aoms: Enum.count(aoms),
+      nb_aoms_with_data: Enum.count(aoms_with_datasets),
+      nb_regions: Enum.count(regions),
+      nb_regions_completed: regions |> Enum.count(fn r -> r.is_completed end),
+      population_totale: get_population(aoms),
+      population_couverte: get_population(aoms_with_datasets),
+      ratio_aom_with_at_most_warnings: ratio_aom_with_at_most_warnings(aoms_max_severity, total_aom_with_datasets),
+      ratio_aom_good_quality: ratio_aom_good_quality(aoms_max_severity, total_aom_with_datasets),
+      aom_with_errors: Map.get(aoms_max_severity, "Error", 0),
+      aom_with_fatal: Map.get(aoms_max_severity, "Fatal", 0),
+      nb_officical_realtime: nb_officical_realtime(),
+      nb_unofficical_realtime: nb_unofficical_realtime(),
+      nb_reusers: nb_reusers(),
+      nb_reuses: nb_reuses(),
+      nb_dataset_types: nb_dataset_types(),
+      nb_gtfs: count_dataset_with_format("GTFS"),
+      nb_netex: count_dataset_with_format("NeTEx"),
+      nb_bss_datasets: count_dataset_with_format("gbfs"),
+      nb_bikes_datasets: nb_bikes()
+    }
+  end
+
+  defp get_population(datasets) do
+    datasets
+    |> Enum.reduce(0, &(&1.population + &2))
+    |> Kernel./(1_000_000)
+    |> Kernel.round()
+  end
+
+  defp nb_officical_realtime do
+    rt_datasets =
+      from(d in Dataset,
+        where: d.has_realtime and d.is_active and d.type == "public-transit"
+      )
+
+    Repo.aggregate(rt_datasets, :count, :id)
+  end
+
+  @spec nb_bikes() :: integer
+  defp nb_bikes do
+    bikes_datasets =
+      from(d in Dataset,
+        where: d.type == "bike-sharing"
+      )
+
+    Repo.aggregate(bikes_datasets, :count, :id)
+  end
+
+  defp nb_unofficical_realtime do
+    Enum.count(CSVDocuments.real_time_providers())
+  end
+
+  defp nb_dataset_types do
+    Dataset
+    |> select([d], count(d.type, :distinct))
+    |> Repo.one()
+  end
+
+  defp nb_reusers do
+    Enum.count(CSVDocuments.reusers())
+  end
+
+  defp nb_reuses do
+    Repo.aggregate(Dataset, :sum, :nb_reuses)
+  end
+
+  defp count_dataset_with_format(format) do
+    Resource
+    |> select([r], count(r.dataset_id, :distinct))
+    |> where([r], r.format == ^format)
+    |> Repo.one()
+  end
+
+  @spec compute_aom_max_severity() :: %{binary() => integer()}
+  defp compute_aom_max_severity do
+    # consolidate the maximum error for the current dataset for each dataset
```

Comment:
c'est juste "for each dataset" que l'on veut j'imagine ?

Reply:
à mon avis c'est "for the current datasets of each AOMs" :grin: